### PR TITLE
purge existing panic operations

### PR DESCRIPTION
**Pull request by sanxun0325**
### Describe what this PR does / why we need it
A panic operation will cause the program to exit

### Does this pull request fix one issue?
fixes:https://github.com/alibaba/sentinel-golang/issues/278

### Describe how to verify it
go test ./...

### Special notes for reviews

### CODE REVIEW

File: core/stat/base/bucket_leap_array.go [hunk 1]
```diff
@@ -32,7 +31,8 @@ func (bla *BucketLeapArray) ResetBucketTo(bw *BucketWrap, startTime uint64) *Buc
 // intervalInMs is the time length of sliding window
 func NewBucketLeapArray(sampleCount uint32, intervalInMs uint32) *BucketLeapArray {
 	if intervalInMs%sampleCount != 0 {
-		panic(fmt.Sprintf("Invalid parameters, intervalInMs is %d, sampleCount is %d.", intervalInMs, sampleCount))
+		logging.Error(errors.Errorf("Invalid parameters, intervalInMs is %d, sampleCount is %d.", intervalInMs, sampleCount), "")
```
Comment (sczyh30) [inline core/stat/base/bucket_leap_array.go:34]: Maybe returning an error is better?

File: core/stat/base/leap_array.go [hunk 2]
```diff
@@ -6,6 +6,8 @@ import (
 	"sync/atomic"
 	"unsafe"
 
+	"github.com/alibaba/sentinel-golang/logging"
+
```
Comment (louyuting) [inline core/stat/base/leap_array.go:10]: fmt imports by "go imports"

File: core/stat/base/metric_bucket.go [hunk 3]
```diff
@@ -1,9 +1,11 @@
 package base
 
 import (
-	"fmt"
 	"sync/atomic"
 
+	"github.com/alibaba/sentinel-golang/logging"
+	"github.com/pkg/errors"
+
```
Comment (louyuting) [inline core/stat/base/metric_bucket.go:8]: fmt imports by "go imports"

File: core/stat/base_node.go [hunk 4]
```diff
@@ -19,7 +21,11 @@ type BaseStatNode struct {
 }
 
 func NewBaseStatNode(sampleCount uint32, intervalInMs uint32) *BaseStatNode {
-	la := sbase.NewBucketLeapArray(config.GlobalStatisticSampleCountTotal(), config.GlobalStatisticIntervalMsTotal())
+	la, err := sbase.NewBucketLeapArray(config.GlobalStatisticSampleCountTotal(), config.GlobalStatisticIntervalMsTotal())
+	if err != nil {
+		logging.Error(err, "failed to get BucketLeapArray")
+		return &BaseStatNode{}
```
Comment (louyuting) [inline core/stat/base_node.go:27]: It doesn't make sense.
We might need to handle this error.

File: core/stat/base/bucket_leap_array.go [hunk 5]
```diff
@@ -30,10 +29,9 @@ func (bla *BucketLeapArray) ResetBucketTo(bw *BucketWrap, startTime uint64) *Buc
 
 // sampleCount is the number of slots
 // intervalInMs is the time length of sliding window
+// sampleCount and intervalInMs must be positive and intervalInMs%sampleCount == 0,
+// the validation must be done before call NewBucketLeapArray
```
Comment (luckyxiaoqiang) [inline core/stat/base/bucket_leap_array.go:33]: How about change the param to `bucket span time` and `bucket count` in future？ Is it more easy to use?

Comment (louyuting) [reply core/stat/base/bucket_leap_array.go:33]: Is your meaning bucket count and bucketLengthMs?

Comment (luckyxiaoqiang) [reply core/stat/base/bucket_leap_array.go:33]: Yes.

Comment (louyuting) [reply core/stat/base/bucket_leap_array.go:33]: We could file another PR to enhance here

Comment (luckyxiaoqiang) [reply core/stat/base/bucket_leap_array.go:33]: Please new a issue and assign to me.

Comment (louyuting) [reply core/stat/base/bucket_leap_array.go:33]: You can submit a issue.

**Review by louyuting — APPROVED** [review]
Thanks for nice work

**Pull request merged**